popups/cloud-games/popup.js

```diff
@@ -57,38 +161,51 @@ export default async ({ addon, msg, safeMsg }) => {
         link.target = "_blank";
         link.href = chrome.runtime.getURL("/webpages/settings/index.html#addon-cloud-games");
         link.textContent = msg("addon-settings");
-        return safeMsg("change-studio", {
+        return safeMsg("change-display", {
           settings: link.outerHTML,
         });
       },
+      addFromSelectedTab() {
+        this.addButtonUsed = true;
+        const { id, type } = extractUrl(this.selectedTabUrl);
+        const url = `https://scratch.mit.edu/${type}s/${id}`;
+        addon.popup.changeSettings({
+          displayedGames: [...addon.settings.get("displayedGames"), { url }],
+        });
+        setTimeout(() => location.reload(), 1500);
+      },
     },
     async created() {
       document.title = msg("popup-title");
-      if (shouldFailEarly) return;
-      let res;
+      addon.popup.getSelectedTabUrl().then((url) => {
+        this.selectedTabUrl = url;
+      });
+      let projects;
       try {
-        res = await fetch(`https://api.scratch.mit.edu/studios/${studioId}/projects/?limit=40`);
+        projects = await strategy(addon, displayedGames);
       } catch (e) {
-        console.warn("Error when fetching studios: ", e);
-        this.error = "server-error";
-        return;
-      }
-      if (res.status >= 500) {
-        this.error = "server-error";
-        return;
-      }
-      if (res.status >= 400) {
-        this.error = "general-error";
-        return;
+        if (e instanceof HTTPError) {
+          const code = e.code;
+          if (code >= 500) this.error = "server-error";
+          else if (code >= 400) this.error = "general-error";
+          return;
+        }
+        throw e;
       }
-      const projects = await res.json();
       if (projects.length === 0) {
         this.error = "no-projects";
         return;
       }
-      // TODO: add currently opened game to projects array. Sort function should put it on top
```

Comment:
Can you add this feature? (Check if online users includes current user name, if yes move it to top, and add visual effect for example add orange border)

Reply:
Not for now